blockchain/types/transaction.go

```diff
@@ -456,18 +490,26 @@ func (tx *Transaction) AsMessageWithAccountKeyPicker(s Signer, picker AccountKey
 // WithSignature returns a new transaction with the given signature.
 // This signature needs to be formatted as described in the yellow paper (v+27).
 func (tx *Transaction) WithSignature(signer Signer, sig []byte) (*Transaction, error) {
-	r, s, v, err := signer.SignatureValues(sig)
+	r, s, v, err := signer.SignatureValues(tx, sig)
 	if err != nil {
 		return nil, err
 	}
 	cpy := &Transaction{data: tx.data}
+
+	if tx.Type().IsEthTypedTransaction() {
+		te, ok := cpy.data.(TxInternalDataEthTyped)
+		if ok {
+			te.setSignatureValues(signer.ChainID(), v, r, s)
+		}
```

Comment:
I think we need `return te, nil` here and also have to add `if !ok` handling

Reply:
I agree with jamie.
We should add return statement in the `if ok` statement and it would be great to add error handling when conversion to `TxInternalDataEthTyped` is failed.

Reply:
I defined errNotImplementTxInternalEthTyped and apply it. Would you check?